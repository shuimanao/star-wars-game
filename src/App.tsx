import React, { FC, useState } from 'react';
import { Button, Stack } from '@mui/material';
import { getPeople, getStarships } from './api/fetchData';
import { PlayPeople } from './components/PlayPeople';
import { PlayStarships } from './components/PlayStarships';
import { AppProps, GameState, SwapiResult } from './types';

export const App: FC<AppProps> = () => {
  const [loading, setLoading] = useState(false);
  const [people, setPeople] = useState<SwapiResult>([]);
  const [starships, setStarships] = useState<SwapiResult>([]);
  const [gameState, setGameState] = useState<GameState>({ status: 'initial' });

  const handlePeopleButtonClick = async () => {
    setLoading(true);
    const people = await getPeople();
    people && setPeople(people.results);
    setLoading(false);
    setGameState({ status: 'in_progress', chosenResource: 'people' });
  };
  const handleStarshipsButtonClick = async () => {
    setLoading(true);
    const starships = await getStarships();
    starships && setStarships(starships.results);
    setLoading(false);
    setGameState({ status: 'in_progress', chosenResource: 'starships' });
  };

  return (
    <Stack justifyContent="center" alignItems="center" spacing="20px">
      {gameState.status === 'initial' && (
        <Stack justifyContent="center" alignItems="center" spacing="20px">
          <p>Choose resource to play:</p>
          <Button
            variant="contained"
            onClick={handlePeopleButtonClick}
            data-testid="choose-people-btn"
          >
            People
          </Button>
          <Button
            variant="contained"
            onClick={handleStarshipsButtonClick}
            data-testid="choose-starships-btn"
          >
            Starships
          </Button>
        </Stack>
      )}
      {loading && <div>Loading data...</div>}
      {gameState.status === 'in_progress' && (
        <>
          {gameState.chosenResource === 'people' && (
            <PlayPeople data={people} setGameState={setGameState} />
          )}
          {gameState.chosenResource === 'starships' && (
            <PlayStarships data={starships} setGameState={setGameState} />
          )}
        </>
      )}
    </Stack>
  );
};
