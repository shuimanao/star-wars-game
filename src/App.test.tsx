import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { screen } from '@testing-library/dom';
import '@testing-library/jest-dom';
import { App } from './App';

test('<App /> renders correctly', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();

  const element = screen.getByTestId('choose-people-btn');
  userEvent.click(element);

  expect(screen.queryAllByTestId('people-chosen')).toBeTruthy();
});
