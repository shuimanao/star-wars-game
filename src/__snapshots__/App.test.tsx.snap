// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div>
  <div
    class="css-1i4si3b-MuiStack-root"
  >
    <div
      class="css-1i4si3b-MuiStack-root"
    >
      <p>
        Choose resource to play:
      </p>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        data-testid="choose-people-btn"
        tabindex="0"
        type="button"
      >
        People
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        data-testid="choose-starships-btn"
        tabindex="0"
        type="button"
      >
        Starships
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
